src/lua/api-gateway/dogstatsd/Dogstatsd.lua

```diff
@@ -0,0 +1,106 @@
+
+--- Auxiliary module used to extract dogstats deamon initialization methods, incrementation of metrics
+-- and calls to other useful functions
+-- employed to measure calls to the Oauth provider: number, duration, etc.
+
+local Dogstatsd = {}
+
+function Dogstatsd:new(o)
+    o = o or {}
+    setmetatable(o, self)
+    self.__index = self
+    return o
+end
+
+--- Loads a lua gracefully. If the module doesn't exist the exception is caught, logged and the execution continues
+-- @param module path to the module to be loaded
+-- @return The loaded module, or nil if the module cannot be loaded
+--
+local function loadrequire(module)
+    ngx.log(ngx.DEBUG, "Loading module [" .. tostring(module) .. "]")
+    local function requiref(module)
+        return require(module)
+    end
+
+    local status, result = pcall(requiref, module)
+    if not (status) then
+        ngx.log(ngx.WARN, "Could not load module [", module, "]. Error ", result)
+        return nil
+    end
+
+    return result
+end
+
+local dogstatsd
+
+--- Returns an instance of dogstatsd only if it does not already exist. Returns the instance if the feature is enabled
+-- @param none
+-- @return An instance of dogstatsd or nil if the class cannot be instantiated
+--
+local function getDogstatsd()
+    local dogstatsHost = ngx.var.dogstatsHost
+    if ngx.var.isDogstatsEnabled == nil or ngx.var.isDogstatsEnabled == "false" then
```

Comment:
use a local variable

Reply:
This is Alex's code, but sure 👍 will do.